src/main/java/me/botsko/prism/wands/InspectorWand.java

```diff
@@ -1,137 +1,112 @@
-				params.addSpecificBlockLocation(sibling.getLocation());
-			}
-
-			// Ignoring any actions via config?
-			if (params.getActionTypes().size() == 0) {
-				@SuppressWarnings("unchecked") final ArrayList<String> ignoreActions = (ArrayList<String>) plugin.getConfig()
-						.getList("prism.wands.inspect.ignore-actions");
-				if (ignoreActions != null && !ignoreActions.isEmpty()) {
-					for (final String ignore : ignoreActions) {
-						params.addActionType(ignore, MatchRule.EXCLUDE);
-					}
-				}
-			}
-			boolean timeDefault = false;
-			for (final String _default : params.getDefaultsUsed()) {
-				if (_default.startsWith("t:")) {
-					timeDefault = true;
-				}
-			}
-			if (timeDefault) {
-				params.setIgnoreTime(true);
-			}
-
-			// Query
-			final ActionsQuery aq = new ActionsQuery(plugin);
-			final QueryResult results = aq.lookup(params, player);
-			if (!results.getActionResults().isEmpty()) {
-
-				final String blockname = Prism.getItems().getAlias(block.getType(), block.getBlockData());
-				player.sendMessage(Prism.messenger.playerHeaderMsg(ChatColor.GOLD + "--- Inspecting " + blockname
-						+ " at " + loc.getBlockX() + " " + loc.getBlockY() + " " + loc.getBlockZ() + " ---"));
-				if (results.getActionResults().size() > 5) {
-					player.sendMessage(Prism.messenger.playerHeaderMsg("Showing " + results.getTotalResults()
-							+ " results. Page 1 of " + results.getTotal_pages()));
-				}
-				for (final me.botsko.prism.actions.Handler a : results.getPaginatedActionResults()) {
-					final ActionMessage am = new ActionMessage(a);
-					if (parameters.hasFlag(Flag.EXTENDED)
-							|| plugin.getConfig().getBoolean("prism.messenger.always-show-extended")) {
-						am.showExtended();
-					}
-					player.sendMessage(Prism.messenger.playerMsg(am.getMessage()));
-				}
-			} else {
-				final String space_name = (block.getType().equals(Material.AIR) ? "space"
-						: block.getType().toString().replaceAll("_", " ").toLowerCase()
-						+ (block.getType().toString().endsWith("BLOCK") ? "" : " block"));
-				player.sendMessage(Prism.messenger.playerError("No history for this " + space_name + " found."));
-			}
-		});
-	}
-
-	/**
-	 * 
-	 */
-	@Override
-	public void playerRightClick(Player player, Entity entity) {
-	}
+    /**
+     * Constructor.
+     * @param plugin Prism
+     */
+    public InspectorWand(Prism plugin) {
+        super(plugin);
+    }
+
+
+    @Override
+    public void playerRightClick(Player player, Entity entity) {
+    }
+
+    @Override
+    public void playerLeftClick(Player player, Location loc) {
+        showLocationHistory(player, loc);
+    }
+
+    @Override
+    public void playerRightClick(Player player, Location loc) {
+        showLocationHistory(player, loc);
+    }
+
+    /**
+     * Show the Location History for a player.
+     *
+     * @param player Player
+     * @param loc    Location
+     */
+    private void showLocationHistory(final Player player, final Location loc) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The method 'showLocationHistory(Player, Location)' has an NPath complexity of 220](https://app.codacy.com/gh/AddstarMC/Prism-Bukkit/pullRequest?prid=5438808)